### ISSUE TITLE
[Bug] Configurations should be bump according to the max bump of the dependencies they has.

### ISSUE THREAD

**Issue by sarvaje** [webhintio/hint#2736]
Maybe I'm wrong, but I thing that if we have a breaking change in one of a configuration dependency, it version should be updated also as a breaking change.

**Comment by molant** [comment]
I believe there were some discussion about this in the release epic.

From the top of my mind:
* if change breaks hintrc then it has to be a manual commit and will have whatever the dev puts ("Breaking" hopefully)
* new package is minor even though it's a major in the package when is first released
* breaking in the hint that doesn't change the hintrc doesn't really change the api to consume it so we consider it patch. This is the most controversial one but kind of fit in the semver logic and prevented us from shipping new versions all the time and having possible issues getting the latest one

That said, open to revisit this is.


________________________________

**Comment by antross** [comment]
> if change breaks hintrc then it has to be a manual commit and will have whatever the dev puts ("Breaking" hopefully)

While this makes sense to me, wouldn't most of what we'd tag as Breaking for a hint affect `.hintrc` in some way? Maybe we should always propagate Breaking changes for a hint through to configurations and leave the rest as-is. Or if we can come up with some way to automate identifying the "affects hintrc" case that'd be cool too.

My concern with leaving this fully manual is that the only files affected will be in the hint, so it will be easy to forget to manually bump the versions of configurations that depend on it.

**Comment by molant** [comment]
My reasoning is that if the ".hintrc" doesn't change dramatically (I.e. There's a change in the schema and it is no longer compatible with a previous version) then it is not a breaking change.


________________________________

**Comment by antross** [comment]
> My reasoning is that if the ".hintrc" doesn't change dramatically (I.e. There's a change in the schema and it is no longer compatible with a previous version) then it is not a breaking change.

Makes sense to me. So it seems we should update the release script to treat breaking changes in hints as a breaking change for any associated configurations. But I don't think we need special handling for other types of bumps as the title of this issue suggests.

@molant and @sarvaje Does this sound good to both of you?

**Comment by molant** [comment]
That's seems right. If there's a breaking on a hint then we've probably changed the reports and the output of the config changes.

My only concern is that we could  bump the configs quite a lot and that will force us to release new versions of the hint package...

Do we have an idea of the next breaking changes and how many we've done in the past to have a better idea?


________________________________

**Comment by antross** [comment]
> My only concern is that we could  bump the configs quite a lot and that will force us to release new versions of the hint package...

With the current release script this technically already happens. We get a patch update to the associated configuration which causes a patch update to `hint` every time we update any hint 🙄

Technically it's getting a bit better with our next updates because I throttled `optionalDependencies` to only update on major releases of dependencies (which is how `hint` references our default configurations). Still I expect Dependabot will probably step in and try to update them anyway between releases...

**Comment by molant** [comment]
@antross didn't you end up implementing this? Can it be closed?

**Comment by antross** [comment]
No, I haven't added anything to make configurations bump according to the max bump of their dependencies. Currently they'll still only have a patch bump regardless of what type of a bump referenced hints have.

**Comment by molant** [comment]
If we land #2968 in multiple PRs and have to publish several times we might want to hold before implementing this 🙃
